app/controllers/api/v0/articles_controller.rb

```diff
@@ -74,7 +74,12 @@ def update
       private
 
       def set_user
-        @user = current_user
+        if params[:key]
+          @api_secret = ApiSecret.find_by_secret(params[:key]) || not_found
```

Comment:
Unless `has_secure_token` does something fancy with finders, this will be susceptible to [timing attack](https://en.wikipedia.org/wiki/Timing_attack). Instead the find should happen on the ID and then use [secure_compare](https://stackoverflow.com/questions/22747052/how-to-use-rails-secure-compare) for the secret.

See also: https://hackernoon.com/timing-based-blind-sql-attacks-bd276dc618dd

Reply:
Thanks for the heads up.

I now also realize this won't be functional anyway. I should have put the WIP on this because I mostly uploaded this because my local tests were hanging for some reason.

Reply:
I'd like to add a consideration for when you pick up this PR again: although the secret key in the URL params is perfectly valid I'd recommend against it for a very simple reason: it ends up in logs, in the occasional shared URL and so on. 

A possible alternative is to use a custom header like: `ApiKey: abcdef123456`

* It could be `X-ApiKey` but X- headers are [deprecated](https://tools.ietf.org/html/rfc6648)
* Using the `Authorization` header is another alternative but since it's a standard header it means it needs to be overloaded and can complicate client's code if used for multiple things (what if in the future you want to add OAuth to the API?)

Adding a header adds no cost to the client